### PR TITLE
Fix/composite header icon prop

### PR DESCRIPTION
**Pull request by Merkur39**
The "icon" property can be undefined.
Therefore, you must make sure that it is defined before any processing.

I also take the opportunity to make a refactor :)

### CODE REVIEW

File: packages/cozy-mespapiers-lib/src/components/CompositeHeader/CompositeHeaderImage.spec.js [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+      text="Text"
+    />
+  )
+}
+
+describe('CompositeHeaderImage', () => {
+  it('should return nothing if icon & fallbackIcon props is undefined', () => {
+    const { container } = setup()
+
+    expect(container).toMatchInlineSnapshot('<div />')
```
Comment (trollepierre) [inline packages/cozy-mespapiers-lib/src/components/CompositeHeader/CompositeHeaderImage.spec.js:22]: nit: don't we use https://github.com/testing-library/jest-dom ?

to be honest, that test would be better with a https://github.com/testing-library/jest-dom#tobeemptydomelement=
```suggestion
    expect(container).toBeEmptyDOMElement()
```

File: packages/cozy-mespapiers-lib/src/components/CompositeHeader/CompositeHeaderImage.jsx [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+  if (!icon && !fallbackIcon) {
+    return null
+  }
+
+  const iconName = icon && icon.split('.')[0]
+  const src = images[iconName] || fallbackIcon
+  const isBitmap = typeof src === 'string' && src.endsWith('.png')
+
+  if (isBitmap) {
+    return <img data-testid={src} src={src} alt="illustration" />
```
Comment (trollepierre) [inline packages/cozy-mespapiers-lib/src/components/CompositeHeader/CompositeHeaderImage.jsx:80]: nit: should we test that part?

**Review by trollepierre — APPROVED** [review]
good job in LGTM 🎉

File: packages/cozy-mespapiers-lib/src/components/CompositeHeader/CompositeHeaderImage.spec.js (hunk 1 above)
Comment (Merkur39) [reply packages/cozy-mespapiers-lib/src/components/CompositeHeader/CompositeHeaderImage.spec.js:22]: En effet, merci j'étais passé à côté 👍

**Comment by Merkur39** [comment]
[Certaines qualifications de documents font crash l'app](https://trello.com/c/1qg6lSed/881-certaines-qualifications-de-documents-font-crash-lapp)

**Pull request merged**